src/main/java/org/embulk/input/zendesk/utils/ZendeskUtils.java

```diff
@@ -1,150 +1,88 @@
-                        pageBuilder.setNull(column);
-                    }
-                    else {
-                        pageBuilder.setTimestamp(column, timestamp);
-                    }
-                    return null;
-                });
-            }
-
-            @Override
-            public void booleanColumn(Column column)
-            {
-                JsonNode data = record.get(column.getName());
-
-                setColumn(column, data, (value) -> {
-                    pageBuilder.setBoolean(column, value.asBoolean());
-                    return null;
-                });
-            }
-
-            @Override
-            public void longColumn(Column column)
-            {
-                JsonNode data = record.get(column.getName());
-
-                setColumn(column, data, (value) -> {
-                    pageBuilder.setLong(column, value.asLong());
-                    return null;
-                });
-            }
-
-            @Override
-            public void doubleColumn(Column column)
-            {
-                JsonNode data = record.get(column.getName());
+        final JsonNode node = ZendeskUtils.parseJsonNode(jsonText);
+        if (node.isObject()) {
+            return (ObjectNode) node;
+        }
 
-                setColumn(column, data, (value) -> {
-                    pageBuilder.setDouble(column, value.asDouble());
-                    return null;
-                });
-            }
+        throw new DataException("Expected object node to parse but doesn't get");
+    }
 
-            private void setColumn(Column column, JsonNode data, Function<JsonNode, Void> setter)
-            {
-                if (isNull(data)) {
-                    pageBuilder.setNull(column);
-                    return;
-                }
-                setter.apply(data);
-            }
-        });
-        pageBuilder.addRecord();
+    public static Iterator<JsonNode> getListRecords(final JsonNode result, final String targetJsonName)
+    {
+        if (!result.has(targetJsonName) || !result.get(targetJsonName).isArray()) {
+            throw new DataException(String.format("Missing '%s' from Zendesk API response", targetJsonName));
+        }
+        return result.get(targetJsonName).elements();
     }
 
-    private static boolean isNull(JsonNode jsonNode)
+    public static boolean isNull(final JsonNode jsonNode)
     {
         return jsonNode == null || jsonNode.isNull();
     }
 
-    /*
-     * For getting the timestamp value of the node
-     * Sometime if the parser could not parse the value then return null
-     * */
-    private static Timestamp getTimestampValue(String value)
+    public static URIBuilder getURIBuilder(final String urlString)
+    {
+        final URI uri;
+        try {
+            uri = new URI(urlString);
+        }
+        catch (final URISyntaxException e) {
+            throw new ConfigException("Login url is invalid format " + urlString);
```

Comment:
The exception message does not fit with the name of method, method name is general for URI, exception is specific to `login_url` only.

Reply:
fixed